spec/unit/face/config_spec.rb

```diff
@@ -32,14 +32,42 @@ module PuppetFaceSpecs
     expect { subject.print("syslogfacility", :section => "user") }.to have_printed('file')
   end
 
-  it "defaults to all when no arguments are given" do
+  it "does not print a warning when a section is given" do
+    Puppet.settings.parse_config(<<-CONF)
+    [user]
+    syslogfacility = file
+    CONF
+
+    Puppet.expects(:warning).never
+    subject.expects(:report_section_and_environment).with('user', 'production')
+
+    expect { subject.print("syslogfacility", :section => "user") }.to have_printed('file')
+    end
+
+  it "does print a warning when a section is not given" do
+    Puppet.settings.parse_config(<<-CONF)
+    [main]
+    syslogfacility = elif
+    [user]
+    syslogfacility = file
+    CONF
+
+    Puppet.expects(:warning).once
+    subject.expects(:report_section_and_environment).with('main', 'production')
+
+    expect { subject.print("syslogfacility") }.to have_printed('elif')
+  end
+
+  it "when no arguments are given" do
     subject.expects(:puts).times(Puppet.settings.to_a.length)
+    subject.expects(:report_section_and_environment)
 
     subject.print
   end
 
-  it "prints out all of the settings when asked for 'all'" do
+  it "when asked for 'all'" do
```

Comment:
Curious about the test name change. Seems to read more like a context now than as a test.

Reply:
Yeah, this was an edit error.  I had converted a couple of tests to be in a context and it was giving me weird results, so I changed it back but I missed this string.  Thanks!